tfsdk/resource_identity.go

```diff
@@ -70,6 +69,16 @@ func (s *ResourceIdentity) Set(ctx context.Context, val interface{}) diag.Diagno
 //
 // Lists can only have the next element added according to the current length.
 func (s *ResourceIdentity) SetAttribute(ctx context.Context, path path.Path, val interface{}) diag.Diagnostics {
+	// If s is nil, then calling s.data triggers a nil pointer error so we return the error diag here
+	if s == nil {
+		return diag.Diagnostics{
+			diag.NewErrorDiagnostic(
+				"Unexpected Move State Response",
+				"An unexpected error was encountered when creating the move state response. Identity data was returned by the provider move state operation, but the resource does not indicate identity support.\n\n"+
+					"This is always a problem with the provider and should be reported to the provider developer."),
+		}
```

Comment:
I think this is a good idea in combination with the other suggestions I left, especially since resource identity frequently uses pointers. My only notes would be to adjust the message and add similar errors to the other methods on `tfsdk.ResourceIdentity`! `Get`, `GetAttribute`, `PathMatches`, `Set` and `SetAttribute`

```suggestion
		return diag.Diagnostics{
			diag.NewErrorDiagnostic(
				"Missing Identity Definition",
				"An unexpected error was encountered when attempting to set a resource identity attribute. The resource does not indicate support via a resource identity schema.\n\n"+
					"This is always a problem with the provider and should be reported to the provider developer."),
		}
```